argo-workflows/Makefile

```diff
@@ -4,7 +4,9 @@ app.yaml:
 	@kubectl create --dry-run=client ns -o yaml argo >> app.yaml
 	@echo --- >> app.yaml
 	@kubectl apply --dry-run=client -n argo -o yaml -f https://raw.githubusercontent.com/argoproj/argo-workflows/stable/manifests/install.yaml >> app.yaml
-	@sed -i '' -e '/^version:/d' manifest.yaml
+	@echo --- >> app.yaml
+	@kubectl -n argo create configmap workflow-controller-configmap --dry-run=client -o yaml --from-file=config >> app.yaml
```

Comment:
Shouldn't the path for --from-file be  `https://raw.githubusercontent.com/argoproj/argo-workflows/config` ?

Reply:
Hi @saiyam1814 

I'm not sure how that will work because there currently isn't any key-value pair config file available in the official Github repo of Argo workflows. That's why the file `config` with custom configuration was committed so that it can be used to generate `app.yaml` using the `make` command.

Reply:
I need to see how `--from-file=config` will work on the boxes.

Reply:
Without `--from-file=config`, here's how the configmap will look like:
```bash
$ kubectl -n argo describe cm workflow-controller-configmap
Name:         workflow-controller-configmap
Namespace:    argo
Labels:       <none>
Annotations:  <none>

Data
====

BinaryData
====

Events:  <none>
$ 
```

After passing the `config`, here's what the configmap looks like:
```bash
$ kubectl -n argo describe cm workflow-controller-configmap
Name:         workflow-controller-configmap
Namespace:    argo
Labels:       <none>
Annotations:  <none>

Data
====
config:
----
containerRuntimeExecutor: pns


BinaryData
====

Events:  <none>
$
```

The impact on jobs(pods) without and with this configuration change has been shared in the first comment.

If I've failed to answer your query then apologies, I may have not totally understood your question.

Reply:
Btw, `config` is a file that was introduced as part of this commit

Reply:
Looping in the maintainer @protosam for guidance.

Reply:
@protosam what are your thoughts on this PR?

Reply:
Will followup on this over this coming weekend.

Reply:
Using `--from-file=config` is fine. The Makefile is to pre-generate `app.yaml` before commit. The working directory for Makefiles will always be the directory the Makefile is in.

There is a problem though. If you do `kubectl create -f app.yaml` against the pre-generated `app.yaml`, you'll get this error.
```
Error from server (AlreadyExists): error when creating "app.yaml": configmaps "workflow-controller-configmap" already exists
```

The problem is that the install manifest already has a `ConfigMap/workflow-controller-configmap` object. This needs to reduced to a single configmap object.

If you use `kubectl apply`, this isn't an issue, but I don't like that there's two entries for the same ConfigMap.

My only recommendation for change is to introduce using `jq` into the Makefile to filter out the ConfigMap provided from the default manifest. The one that the workflows manifest provides is empty anyways.

```Makefile
build: app.yaml

app.yaml:
	@kubectl create --dry-run=client ns -o yaml argo >> app.yaml
	@echo --- >> app.yaml
	@kubectl -n argo create configmap workflow-controller-configmap --dry-run=client -o yaml --from-file=config >> app.yaml
	@echo --- >> app.yaml
	@kubectl apply --dry-run=client -n argo -o json -f https://raw.githubusercontent.com/argoproj/argo-workflows/stable/manifests/install.yaml | \
		jq 'del(.items[] | \
		select(.metadata.name == "workflow-controller-configmap"))' | \
		kubectl apply --dry-run=client -n argo -o yaml -f - >> app.yaml
	@sed -i -e '/^version:/d' manifest.yaml
	@grep -m1 'image: .*/workflow-controller:' app.yaml | cut -d: -f3 | sed 's/\(.*\)/version: "\1"/' >> manifest.yaml

.PHONY: clean
clean:
	@rm app.yaml
```

Reply:
Hi @protosam,

Thank you for pointing out the problem. As per your suggestion, I've made the correction. 

Kindly review.

Reply:
@saiyam1814 this should be good now. 

This is the workflows documentation for what's being configured as a default for workflows: https://argoproj.github.io/argo-workflows/workflow-executors/#process-namespace-sharing-pns